library/src/main/java/com/novoda/downloadmanager/RoomV2DatabaseFiles.java

```diff
@@ -0,0 +1,23 @@
+package com.novoda.downloadmanager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class RoomV2DatabaseFiles implements V2DatabaseFiles {
+    private final RoomFileDao roomFileDao;
+
+    RoomV2DatabaseFiles(RoomFileDao roomFileDao) {
+        this.roomFileDao = roomFileDao;
+    }
+
+    @Override
+    public List<String> databaseContents() {
```

Comment:
I understand the naming here but I think this is wrong. We are leaking that the backing data-structure is a database.